<!--
	Sample launch file for depthimage_to_laserscan
	As of September 2nd 2021.
-->

<launch>

  <!-- Depth Image to Laser Scan Node -->
  <node name="depthimage_to_laserscan" pkg="depthimage_to_laserscan" type="depthimage_to_laserscan" args="">

    <remap from="image"       to="/zed2/zed_node/depth/depth_registered"/> <!-- change here for your camera depth topic name. Default: "/camera/depth/image_raw" -->
    <remap from="scan" to="dp2ls/scan"/>
    <remap from="camera_info" to="/zed2/zed_node/depth/camera_info"/> 
    <!-- the remap for camera_info by default uses the same path as the one given to the image topic. Default: "<path_to_image_topic>/camera_info>" -->

    <!-- PARAMETERS: -->
    <!-- These are the default values. --> 
    <param name="scan_height"     type="int"    value="4"/> <!-- default: 1 pixel. Number of pixel rows used to generate laser scan. -->
    <param name="scan_time"       type="double" value="0.01"/> <!-- default:0.033, 30 FPS . Time between scans. -->
    <param name="range_min"       type="double" value="0.1"/> <!--default:0.45m. Ranges less than this are considered -Inf. -->
    <param name="range_max"       type="double" value="3.0"/> <!--default: 10m. Ranges less than this are considered +Inf. -->
    <param name="output_frame_id" type="str"    value="base_footprint"/> <!--default: camera_depth_frame. Frame id of the laser scan. -->
	
  </node>
	
</launch>